src/backend/strategies.rs

```diff
@@ -1082,44 +1100,80 @@ pub async fn run_strategy_task<'s>(
                         num_asset_lock_proofs_needed
                     );
                     let asset_lock_proof_time = Instant::now();
-                    for i in 0..(num_asset_lock_proofs_needed) {
-                        if let Some(wallet) = wallet_lock.as_mut() {
-                            // TO-DO: separate this into a function for start_identities, top ups, and inserts, because the balances should all be different
-                            match wallet.asset_lock_transaction(
-                                None,
-                                strategy.start_identities.starting_balances,
-                            ) {
-                                Ok((asset_lock_transaction, asset_lock_proof_private_key)) => {
-                                    match try_broadcast_and_retrieve_asset_lock(sdk, &asset_lock_transaction, &wallet.receive_address(), 2).await {
-                                        Ok(asset_lock_proof) => {
-                                            tracing::info!("Successfully obtained asset lock proof number {}", i + 1);
-                                            asset_lock_proofs.push((asset_lock_proof, asset_lock_proof_private_key));
-                                        }
-                                        Err(_) => {
-                                            tracing::error!("Failed to obtain asset lock proof number {} after retries", i + 1);
-                                            return BackendEvent::StrategyError {
-                                                error: format!("Failed to obtain all asset lock proofs. Suggest to try rerunning."),
-                                            };                        
-                                        }
+
+                    // Broadcast asset locks and receive proofs.
+                    let permits = Arc::new(Semaphore::new(20));
+                    let starting_balance = strategy.start_identities.starting_balances;
+                    let processed = Arc::new(AtomicUsize::new(0));
```

Comment:
it doesn't mean much in terms of performance but I'm curious if we had to use Arc both for `permits` and `processed`; do we use multithreaded executor?